analytics/src/main/java/com/segment/analytics/SegmentIntegration.java

```diff
@@ -233,6 +251,42 @@ public void track(TrackPayload track) {
     dispatchEnqueue(track);
   }
 
+  @Override
+  public void reportGoalResult(ReportGoalResultPayload report) {
+    Properties reportProperties = new Properties();
+
+    if (attemptProperties.containsKey(report.event())) {
+      reportProperties.putAll(attemptProperties.remove(report.event()));
```

Comment:
What if attemptProperties.remove(...) returns null?  I.e., if the event has no entry in attemptProperties?